### PR TITLE
Added design meeting notes from 2020-07-24.

### PR DESCRIPTION
**Pull request by cgranade**
This PR adds notes from the 24 July 2020 Q# API Design meeting, including a resolution on the API design proposed in #304.

### CODE REVIEW

File: Design/meetings/2020/api-design-2020-07-24.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        - Q# API design should consider what reasonable type classes may look like.
+
+- API for target machines to expose random seed setting.
+    - For Q# standalone, propagate via project properties?
+    - Inappropriate to expose at Q# level, since not all targets need expose a PRNG at all; could be QRNG-backed.
+    - The proposed API intentionally does not distinguish between an engine that produces randomness, and distributions based on that source of randomness.
+    - **Resolution:** Proposal unmodified, as random seeds are simulator-specific and shouldn't be exposed at Q# level.
+    
+- No distribution over complex numbers in current proposal.
+    - Should we introduce ComplexDistribution?
```
Comment (guenp) [inline Design/meetings/2020/api-design-2020-07-24.md:27]:
```suggestion
    - Should we introduce `ComplexDistribution`?
```

File: Design/meetings/2020/api-design-2020-07-24.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- This proposal requires changes to the QSharpCore project in the qsharp-runtime repo, and thus should be done as soon as possible to help manage dependency.
+    - How difficult would it be to adapt simulation runtime and translators? Should be fine, as runtime changes localized to `SimulatorBase`.
+    - **Resolution:** Create issue on qsharp-runtime to track needed modifications to QSharpCore and `SimulatorBase`.
+
+- Impact on / from Q# language discussions:
+    - Possible language discussion area: type classes.
+        - E.g.: Cannot meaningfully compare equality on callables, so even equality requires type-class–like concept.
+        - Q# API design should consider what reasonable type classes may look like.
+
+- API for target machines to expose random seed setting.
```
Comment (guenp) [inline Design/meetings/2020/api-design-2020-07-24.md:20]: What are target machines? Do you mean quantum hardware?

File: Design/meetings/2020/api-design-2020-07-24.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - For Q# standalone, propagate via project properties?
```
Comment (guenp) [inline Design/meetings/2020/api-design-2020-07-24.md:21]:
```suggestion
    - For Q# standalone implementation, should we propagate the random seed settings via project properties?
```

File: Design/meetings/2020/api-design-2020-07-24.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Q# API Design Meeting / 24 July 2020
+
```
Comment (guenp) [inline Design/meetings/2020/api-design-2020-07-24.md:2]: I would suggest to add handles of attendees here

File: Design/meetings/2020/api-design-2020-07-24.md (hunk 2 above)
Comment (cgranade) [reply Design/meetings/2020/api-design-2020-07-24.md:20]: We've generally used that term to be inclusive both of hardware and of the simulation runtime.

**Comment by cgranade** [comment]
Thanks for the review, @guenp!

**Pull request merged**